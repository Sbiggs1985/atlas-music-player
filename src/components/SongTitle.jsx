import React from 'react';

const SongTitle = () => {
  return (
    <div className="w-400"> 
      <h2 className="text-xl text-black-900 font-bold">Painted in Blue</h2>
      <p className="text-sm text-gray-600">Soul Canvas</p>
    </div>
  );
};

export default SongTitle;