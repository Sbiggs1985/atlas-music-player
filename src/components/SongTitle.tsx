import React from 'react';

// Define props type for the component
interface SongTitleProps {
  title: string;
  artist: string;
}

const SongTitle: React.FC<SongTitleProps> = ({ title, artist }) => {
  return (
    <div className="w-400">
      <h2 className="text-xl text-black-900 font-bold">{title}</h2>
      <p className="text-sm text-gray-600">{artist}</p>
    </div>
  );
};

export default SongTitle;

// Props added: title & artist
// Instead of hardcoded values (painted in blue and Soul Canvas), the title & artist now change dynamcally based on props.
// Defined an interface SongTitleProps to ensure title & aartist are string.