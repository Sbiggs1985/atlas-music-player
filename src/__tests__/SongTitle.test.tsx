import { render, screen } from "@testing-library/react";
import SongTitle from "@/components/SongTitle";
import { describe, it, expect } from "vitest";

describe("SongTitle Component", () => {
  it("renders the song title and artist", () => {
    render(<SongTitle title="Painted in Blue" artist="Soul Canvas" />);

    expect(screen.getByText("Painted in Blue")).toBeInTheDocument();
    expect(screen.getByText("Soul Canvas")).toBeInTheDocument();
  });

  it("matches snapshot with a title and artist", () => {
    const { asFragment } = render(<SongTitle title="Painted in Blue" artist="Soul Canvas" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("matches snapshot with a different title and artist", () => {
    const { asFragment } = render(<SongTitle title="Golden Haze" artist="Velvet Waves" />);
    expect(asFragment()).toMatchSnapshot();
  });
});