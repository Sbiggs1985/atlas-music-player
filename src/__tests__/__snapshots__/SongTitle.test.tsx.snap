// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SongTitle Component > matches snapshot with a different title and artist 1`] = `
<DocumentFragment>
  <div
    class="w-400"
  >
    <h2
      class="text-xl text-black-900 font-bold"
    >
      Golden Haze
    </h2>
    <p
      class="text-sm text-gray-600"
    >
      Velvet Waves
    </p>
  </div>
</DocumentFragment>
`;

exports[`SongTitle Component > matches snapshot with a title and artist 1`] = `
<DocumentFragment>
  <div
    class="w-400"
  >
    <h2
      class="text-xl text-black-900 font-bold"
    >
      Painted in Blue
    </h2>
    <p
      class="text-sm text-gray-600"
    >
      Soul Canvas
    </p>
  </div>
</DocumentFragment>
`;
